flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/security/token/HiveServer2DelegationTokenProvider.java

```diff
@@ -0,0 +1,235 @@
+                (PrivilegedExceptionAction<ObtainedDelegationTokens>)
+                        () -> {
+                            Preconditions.checkNotNull(hiveConf);
+                            Hive hive = Hive.get((HiveConf) hiveConf);
+                            Clock clock = Clock.systemDefaultZone();
+                            try {
+                                LOG.info("Obtaining Kerberos security token for HiveServer2");
+
+                                String principal =
+                                        hiveConf.getTrimmed(
+                                                "hive.metastore.kerberos.principal", "");
+
+                                String tokenStr =
+                                        hive.getDelegationToken(
+                                                UserGroupInformation.getCurrentUser().getUserName(),
+                                                principal);
+                                Token<HiveServer2DelegationTokenIdentifier> hive2Token =
+                                        new Token<>();
+                                hive2Token.decodeFromUrlString(tokenStr);
+
+                                Credentials credentials = new Credentials();
+                                credentials.addToken(hive2Token.getKind(), hive2Token);
+
+                                HiveServer2DelegationTokenIdentifier tokenIdentifier =
+                                        getTokenIdentifier(hive2Token);
+
+                                if (tokenRenewalInterval == null) {
+                                    tokenRenewalInterval =
+                                            getTokenRenewalInterval(
+                                                    clock, tokenIdentifier, hive, tokenStr);
+                                }
+                                Optional<Long> validUntil =
+                                        tokenRenewalInterval.flatMap(
+                                                interval ->
+                                                        getTokenRenewalDate(
+                                                                clock, tokenIdentifier, interval));
+
+                                return new ObtainedDelegationTokens(
+                                        HadoopDelegationTokenConverter.serialize(credentials),
+                                        validUntil);
+
+                            } catch (Exception e) {
+                                LOG.error(
+                                        "Failed to obtain delegation token for {}",
+                                        this.serviceName(),
+                                        e);
+                                throw new FlinkRuntimeException(e);
+                            } finally {
+                                Hive.closeCurrent();
+                            }
+                        });
+    }
+
+    Optional<Long> getTokenRenewalInterval(
```

Comment:
Can we write some comment here, why we do it this way?
Also why not make `getTokenRenewalInterval` private?

Reply:
Because the getTokenRenewalInterval method needs to be verified in the unit test，when set to private, I cannot access this method